src/kvstore/NebulaStore.cpp

```diff
@@ -781,32 +781,41 @@ nebula::cpp2::ErrorCode NebulaStore::ingest(GraphSpaceID spaceId) {
   if (!ok(spaceRet)) {
     return error(spaceRet);
   }
+
   auto space = nebula::value(spaceRet);
+  std::vector<std::thread> threads;
+  nebula::cpp2::ErrorCode code = nebula::cpp2::ErrorCode::SUCCEEDED;
+  LOG(INFO) << "Space " << spaceId << " start ingest.";
   for (auto& engine : space->engines_) {
-    auto parts = engine->allParts();
-    for (auto part : parts) {
-      auto ret = this->engine(spaceId, part);
-      if (!ok(ret)) {
-        return error(ret);
-      }
-
-      auto path = folly::stringPrintf("%s/download/%d", value(ret)->getDataRoot(), part);
-      if (!fs::FileUtils::exist(path)) {
-        LOG(INFO) << path << " not existed";
-        continue;
-      }
+    threads.emplace_back(std::thread([&engine, &code, this, spaceId] {
```

Comment:
I think you could run it in folly Executor to avoid create too many threads.

Reply:
each engine will create a thread so it seems fine

Reply:
using Executor is a good idea thanks